### PR TITLE
Add documentation for the Redis transport

### PR DESCRIPTION
**Pull request by alexander-schranz**
This will add documentation how to configure and using the redis transport with the messenger component. 

https://github.com/symfony/symfony/pull/30917

#EUFOSSA

### CODE REVIEW

File: messenger.rst [hunk 1]
```diff
@@ -209,6 +220,35 @@ This will also configure the following services for you:
...
+.. code-block:: bash
+
+    # .env
+    ###> symfony/messenger ###
+    MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages/group1/consumer1
+    ###< symfony/messenger ###
+
+.. note::
+
+    In order to use Symfony's built-in Redis transport, you will need the Redis
```
Comment (Toflar) [inline messenger.rst]: Can we add the required `composer` command here so lazy people like me can just copy&paste? :P

Comment (alexander-schranz) [reply messenger.rst]: Is this not system specific how the extension is installed? It could be `pecl install redis` or `apt-get` or something else or?

Comment (Toflar) [reply messenger.rst]: Ah I see, misunderstood!

File: messenger.rst [hunk 2]
```diff
@@ -209,6 +220,35 @@ This will also configure the following services for you:
...
 
+Redis
+~~~~~
+
+The redis transport will use the redis streams to queue messages.
+
+.. code-block:: bash
+
+    # .env
+    ###> symfony/messenger ###
```
Comment (OskarStark) [inline messenger.rst]: To me this to flex specific. I would only focus on the ENV var (and the filepath)

File: messenger.rst [hunk 3]
```diff
@@ -209,6 +220,35 @@ This will also configure the following services for you:
...
+    MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages/group1/consumer1
+    ###< symfony/messenger ###
+
+.. note::
+
+    In order to use Symfony's built-in Redis transport, you will need the Redis
+    PHP extension (^4.2), a running Redis Server (^5.0) and the Serializer Component.
```
Comment (OskarStark) [inline messenger.rst]: s/Server/server/

s/Component/component/

**Review by OskarStark — COMMENTED** [review]
Sometimes Redis and sometimes redis is written, we should only use one consistently

File: messenger.rst (hunk 2 above)
Comment (alexander-schranz) [reply messenger.rst]: Thank you!
Think the same. Did remove it also from the amqp configuration example.

File: messenger.rst (hunk 3 above)
Comment (alexander-schranz) [reply messenger.rst]: Fixed, thank you!

**Review by OskarStark — APPROVED** [review]

**Comment by wouterj** [comment]
Congratz on your code merge and many thanks for providing docs @alexander-schranz! I've added the versionadded directive in daaf7b2d6, but apart from that this was perfect.

**Pull request merged**